baggage/baggage.go

```diff
@@ -134,11 +145,11 @@ func (p Property) Value() (string, bool) {
 	return p.value, p.hasValue
 }
 
-// String encodes Property into a string compliant with the W3C Baggage
+// String encodes Property into a header string compliant with the W3C Baggage
 // specification.
 func (p Property) String() string {
 	if p.hasValue {
-		return fmt.Sprintf("%s%s%v", p.key, keyValueDelimiter, p.value)
+		return fmt.Sprintf("%s%s%v", p.key, keyValueDelimiter, valueEscape(p.value))
```

Comment:
Why does this need to be here? W3C baggage format is only one of many possible formats. Baggage package shouldn't do any encoding. Separation of concerns.

Reply:
Backwards compatility is the reason. Otherwise we could remove String methods and Parse function.

Reply:
Up to you if you want to leave it - I think it will keep the code in this split-brain state where baggage package is doing things it's not supposed to be doing, and propagator package is not doing its job.

String() is a pretty bad place to do these things. String() is a method used for debugging, I wouldn't expect it to mangle values. Take the `time` package - it has Parse and Format, which are reasonable counterparts, but they require an encoding parameter.

Reply:
The package is stable and I think we should reduce the number of breaking changes as much as possible.

Reply:
Resolving per https://github.com/open-telemetry/opentelemetry-go/pull/4804#issuecomment-1876601271